### DISCUSSION TITLE
FIL-RetroPGF-1: Nominations for Other

### DISCUSSION THREAD

**Discussion by kkarrancsu** [filecoin-project/community#696] [RetroPGF]
### Other

This is the thread to nominate projects for the **_Other_** category of FIL-RetroPGF-1. As a reminder, this category is intended to capture projects that don't fall into any other categories previously set forth.

### Nomination Template
When nominating a project, please answer the following questions in a comment to this discussion:

1. The project name
2. A description of the positive impact of this project for this specific category over the past six months (Oct 2023 - Mar 2024)
3. A link to the project’s GitHub or Twitter
4. Contact info for the project or project lead

### Nomination Considerations

Things to consider when nominating a project:
- Use your best effort to describe how this project has delivered impact to the Filecoin ecosystem in the previous six months (Oct 2023 - Mar 2024). Use the template provided in each category’s thread when nominating a project.
- Make sure to nominate projects in the relevant category. If you feel that a project spans multiple categories, please choose the one category that you feel it best aligns with. Do not nominate the same project across multiple categories! 
- Note that the organizers of FIL-RetroPGF-1 may choose to reassign a project’s category from its nominated one, if they feel it better aligns elsewhere. 
- Before nominating a project please check if this project has been nominated already, so we can avoid duplication.
RetroPGF is an experiment that Filecoin plans to iterate on regularly. Community feedback is key to helping the process improve with each round. We welcome your notes, reflections, and feedback at any step. Please use this discussion to submit feedback!

Project Nominations will close by Mar 31, 2024.

See [here](https://github.com/filecoin-project/community/discussions/689) for general guidelines on project nomination.

_Note: These threads will be moderated to only include project nominations. Off-topic comments that show support or hype for a project or meta-commentary on the program itself will be removed to keep the thread clean. If you want to show your excitement for a nomination, use the reaction feature!_

**Comment by jnthnvctr** [comment]
The project name:
Dcent

A description of the positive impact of this project for this specific category over the past six months (Oct 2023 - Mar 2024)
Dcent has been the technical counterpart for the Triton team as they have worked on getting the Solana data secured on Filecoin as a part of the Old Faithful project. Dcent has been tip of spear on helping identify and unblock issues that have arisen from this work, and continue to help support this ongoing effort.

The impact was most notably recognized in Q1 2024, when Twitter erupted with enthusiasm that this integration was underway. High profile wins in the community that drive enthusiasm for Filecoin and its utility, and deserve to be rewarded.

A link to the project’s GitHub or Twitter:
https://twitter.com/Dcent_Web3

Contact info for the project or project lead
[hidde@dcent.nl](mailto:hidde@dcent.nl)

**Reply by The-Wayvy** [reply]
I'd support nominating Old Faithful.

But not DCent.

**Reply by jnthnvctr** [reply]
I mean we should submit for triton as well, but dcent did / has been
continuing to do a fair amt of custom work to make this project move

I’m not sure why you’re saying not dcent - they’ve been actually doing a
lot to make this happen

**Reply by anjor** [reply]
I nominated old faithful here btw -- https://github.com/filecoin-project/community/discussions/691#discussioncomment-8931420 

let me know if it should be moved here.

**Reply by herrehesse** [reply]
@The-Wayvy why not us?

**Comment by The-Wayvy** [comment]
1. project name

Raw Filecoin

2. A description of the positive impact of this project for this specific category over the past six months (Oct 2023 - Mar 2024)

Raw Filecoin turns the most popular FIP ever [774](https://github.com/filecoin-project/FIPs/discussions/774) into an app running on the FVM.

3. Link to the project’s GitHub or Twitter

[Github](https://github.com/RawFilecoin/RawFilecoin)

4. Contact info for the project or project lead

[rawfilecoin@gmail.com](mailto:rawfilecoin@gmail.com)

**Reply by RawFilecoin** [reply]
Truly appreciated, @The-Wayvy.

I'd like to elaborate further as Raw Filecoin (RFIL) might be relatively unfamiliar to many in the community. We welcome any suggestions or feedback.

Raw Filecoin (RFIL) is a token distributed to miners on the Filecoin network, aligning with the Filecoin whitepaper's vision. Rewards are allocated based on actual storage contributions, focusing on RawBytes rather than QAP. RFIL is running on FEVM, with voluntary participation from miners who can claim daily rewards.

### RawFilecoin's goals: 

- Restoring the original vision of Filecoin.
- Compensating CC miners without distinction between DC and CC sectors for RFIL rewards.
- Implementing the system in the simplest manner.

For more information, please go to: 
- [RawFilecoin Readme](https://github.com/RawFilecoin/RawFilecoin/blob/main/README.md)
- [Smart Contract](https://filfox.info/en/address/f02952369?t=3)

**Comment by jamescbury** [comment]
Project Name: Confidential Settlements with Embedded Rights (CSER)
[Github](https://github.com/zeroth-tech/CSER)
Project Team: https://zeroth.technology

**Why are we here?**  CSER is a new project, but it is not a new idea... we have been working to solve enterprise supply chain issues using decentralized technology for years now.  There have been a lot of setbacks - but we are getting closer.  As we drive forward cryptographic primitives like CSER there are going to be massive impacts on decentralized storage networks - not necessarily in terms of data volume, but in terms of transactional volume and enterprise users.  These are two things that will drive tremendous value for the Filecoin ecosystem and bring a new ecosystem of players - manufacturers, logistics providers, retailers, consumers, regulators - into the Filecoin community.

**Why is this an "other"?**  CSER is going to be an open-sourced cryptographic technique.  We are in the process right now of writing the white paper which will shortly be followed by a prototype.  A key to our success will be several reference implementations of the technique and we envision that one could run entirely on the FVM.  In this scenario Filecoin would not just be the data component of the solution (which would fall under infrastructure) but it could also be the entire solution (which would fall under product).  Of course we would be happy to place it where it best belongs...

**Why not later?** We realize that impact of CSER is yet to be realized and this makes it difficult to justify a retroPGF, but we believe the value of bringing enterprise supply chain use cases onto the Filecoin network is going to be massive.  Through our work with the DSA we know the barriers for enterprise adoption are still high - solutions such as CSER will lower those barriers because it solves far more than a data storage problem.  We need to get started on developing the reference implementations yesterday...

**Comment by endomorphosis** [comment]
Project Name: IPFS / S3 Huggingface model manager

https://github.com/endomorphosis/ipfs_transformers

Project Team (JusticeDAO)
@endomorphosis 
@coregod360
@mwni 

I initially started this, because I was intending on using the IPFS models for my MLops model delivery, but I started to consider after talking to Yannic Kilcher CEO of Deep Judge, regarding the UX of training models and fetching data from huggingface, that having multiple workers simultaneously all pulling data from huggingface was bad for network congestion, and decided to try to modify the huggingface libraries to fetch models from IPFS, which might include a neighboring node in the same datacenter or a pinning service, or from S3 when there is no pin available, or from HTTPS when neither are available. For those who do not know, machine learning systems need to optimize for latency and bandwidth, in order to keep the GPU/TPU/NPU utilization high. This is why for example instead of a hub and spoke network model, Nvidia and Google use some variant of ring networking called a toroidal network, in order to avoid the bottlenecks in a traditional network layout. This means that each accelerator is directly sharing data with its neighboring accelerators, instead of relying on a centralized router, because if you double your number of accelerators you will double your network throughput per node, rather than halving it with a traditional hub and spoke network layout.

Subsequently the US government is proposing to classify open source models as 'dual use' export controlled, in addition to KYC and deplatforming for people who are 'red flagged'. I have been working on some code to scrape huggingface and serve models on IPFS, mainly out of concerns that a 3d avatar legal secretary will get attacked or shut down by malefactors, because it will be assisting people in their lawsuits against the government, many of whom are being held in jail without a public defender because there are simply none available. However with the new regulations it appears that huggingface will be under pressure to comply with the ITAR regulations, and close access to any machine learning model with more than 10 billion parameters, and will face jail time if they do not comply, so I am creating this to maintain access to ML models even if they are censored. While I certainly think that this regulation is a classic "prior restraint on speech", that will likely take several years to work its way through the courts, and in the meanwhile there needs to be a stopgap measure to mitigate the supply chain risk.

This library acts as a wrapper which adds a static methods .from_auto_download and .from_ipfs to the transformers library, but the repository also contains a scraper for huggingface with a CLI, some model conversion utilities, and a website to easily browse what models are available in the collection. The ipfs_kit library support a wrapper for ipfs_daemon, ipfs_cluster_service and ipfs_cluster_follow, and changing self.role from leecher to worker or master should install / configure it.

Should I receive more funding, I will implement orbitDB to track all of the models / pins, mirror the entire huggingface website and host it using DNS link, create a cache / backup of website and the most popular ML models on hugging face with filecoin contracts to prevent censorship, and implement user side filecoin integration to increase the model delivery speed over the IPFS network which will be useful for their edge compute. I have interviewed 10 people for the role, and have decided that the two additional staff members i would like to hire are @blenderfreaky and @jtainer who have each asked for 80k salary, who will be assigned to work solely on this project, and will assign @coregod360 part time for front end webdev and github pull requests and bug reports.


I am very intent on building AI lawyer technology, and as much as I love IPFS cloud infrastructure, my state of Oregon only has 30% of the required number of public defenders, and the fates of people sitting in jail with no lawyer are more important to me, so I will supervising them and going to work on other parts of the stack. When feasible i will open source parts of the stack that are not relevant to the core business, and have previously open sourced a proof of concept IPFS KNN database, https://github.com/endomorphosis/opendatahack



Sources:
https://www.federalregister.gov/documents/2024/01/29/2024-01580/taking-additional-steps-to-address-the-national-emergency-with-respect-to-significant-malicious

https://www.regulations.gov/document/NTIA-2023-0009-0001

https://venturebeat.com/ai/nist-staffers-revolt-against-potential-appointment-of-effective-altruist-ai-researcher-to-us-ai-safety-institute/

https://yro.slashdot.org/story/24/03/11/185217/us-must-move-decisively-to-avert-extinction-level-threat-from-ai-govt-commissioned-report-says

https://www.opb.org/article/2022/01/20/american-bar-association-finds-oregon-has-just-13-of-needed-public-defenders/

**Comment by mishmosh** [comment]
**Project name:**
IPFS (Interplanetary FileSystem) and IPLD

**Description of the positive impact of this project for this specific category over the past six months (Oct 2023 - Mar 2024):**

This nomination is for the IPFS project as a whole, including protocol stewardship of IPFS, IPLD, and multiformats, outreach, education, and community building. Its impact is woven across the Filecoin project, but for this nomination fall into 2 major buckets:

1. *Building blocks for the Filecoin protocol:*  IPLD, a subproject of IPFS, is the data model of the content addressable web. IPLD's concepts and protocols such as DAG-CBOR, .car file format, and Graphsync are used extensively across the [Filecoin spec](https://spec.filecoin.io/libraries/ipld/) and run in every Filecoin node.

2. *Storage user growth:* IPFS has firmly established CIDs as the default of storage design for Web3. Its 10-year track record as "the backbone of web3" and ongoing new usage contribute to interest, credibility, and client lead generation for Filecoin. Many of Filecoin's storage users cite the efficiency and verifiability of CIDs, alongside cost, as a reason for choosing Filecoin (example: [Victor Chang Cardiac Institute](https://blockworks.co/news/filecoin-cardiac-institute-decentralized-storage)).

**A link to the project’s GitHub or Twitter:**
- https://github.com/ipfs/
- twitter.com/ipfs

**Contact info for the project or project lead:** mosh@ipfs.io on behalf of the IPFS Project

**Reply by endomorphosis** [reply]
I noticed that when trying to index large ML models that the IPFS daemon hashing seems to be single threaded, and therefore somewhat slow when indexing large files. If this is funded, it is my hope that someone in your org can try to create a new spec, to parallelize the hashing of large files.

**Reply by mishmosh** [reply]
Thanks for highlighting this @endomorphosis. I discussed this with @lidel @bumblefudge and others who are heavily involved in IPFS specs. We believe there's room to improve both the spec and implementations to speed up hashing of large files.

Within [kubo](https://github.com/ipfs/kubo/):
- Make core commands like `ipfs dag import|export` and  `ipfs add` as fast as possible (we know they are not)
- Commands like `ipfs add` are not as fast as they "should be", when comparing with `sha256sum` over the number of chunks

Once we have a reference implementation, we can add some rules of thumb how to implement UnixFs hashing and chunking (that may cover block sizes, hashing algorithms, etc.) to "notes for implementers" section of the IPFS [wip Unix specification](https://github.com/ipfs/specs/pull/331).

I created https://github.com/ipfs/kubo/issues/10383 to capture this discussion. Feel free to comment or follow there.

**Comment by kkarrancsu** [comment]
Dear Filecoin Community,

This week marks an important milestone for the FIL-RetroPGF-1 effort.  Nominations for projects closed on Mar 31, 2024.  We received over 150 nominations for projects across the various categories that were eligible for FIL-RetroPGF-1 funding. 

Thank you to the community for taking the time and effort to put these nominations together. The projects nominated reflect all of the incredible work and progress that the Filecoin community have made together.

As FIL-RetroPGF-1 organizers, our next step is to review the nominations and select eligible projects to apply formally to be considered for FIL-RetroPGF-1 funding. We will be using the criteria outlined [in this document](https://fil-retropgf.notion.site/Nomination-Review-Process-32b62a5ea2cd477f83692cbc233edfda) to determine whether a nominated project should be asked to apply.  Nominated projects will be informed soon, and applications are currently slated to open on Apr 8th.

Thank you, stay tuned, and check back to this page and main our [RetroPGF landing page](https://fil-retropgf.notion.site/FIL-RetroPGF-4b6f5358440043c8bb1bf53f0297541e) for future developments!

**Comment by kkarrancsu** [comment]
Dear Filecoin Community,

The FIL-RetroPGF organizers have spent the last week reviewing the nominations submitted in each of the eligible categories. We used the criteria outlined [here](https://fil-retropgf.notion.site/Nomination-Review-Process-32b62a5ea2cd477f83692cbc233edfda) to determine which projects would be invited to apply in this round. Our assessment of eligibility is based on our best interpretation of the information provided in the nomination but tended to bias towards inclusivity, deferring judgement to badgeholders as much as possible. 

The following [page](https://fil-retropgf.notion.site/Round-1-Invited-Applications-3fe95da0c9da4aeba9c6420d1efd0bd6) contains a list of projects that are being invited to apply to be eligible for FIL-RetroPGF-1 funding.

Applications are open from 8 Apr 2024 to 22 Apr 2024. We encourage you to submit your application [here](https://www.fil-retropgf.io/) as soon as possible!

Best regards,
FIL-RetroPGF Organizers

**Comment by kkarrancsu** [comment]
Dear Filecoin Community,

This is a gentle reminder that the application submission deadline is fast approaching. If you are a project invited to apply and have not already done so, please submit your applications [here](https://www.fil-retropgf.io/) by Apr 22.

We have updated the [Round 1 - Invited Applications](https://fil-retropgf.notion.site/Round-1-Invited-Applications-3fe95da0c9da4aeba9c6420d1efd0bd6) page to include a POC. Additionally, we have published [the following guidelines](https://fil-retropgf.notion.site/Project-Application-Guidelines-4cf9df76d0f74da9b02cd33f09cbd5b4) on how to approach the application process and included an FAQ at the bottom.

Best Regards,
FIL-RetroPGF Organizers

**Comment by kkarrancsu** [comment]
Dear Filecoin Community,

We have extended the deadline to submit your applications. If you haven’t already done so, please submit your application [here](http://fil-retropgf.io/) by 12PM UTC Apr 24, 2024. There will be no further extensions.

Best regards.

**Comment by kkarrancsu** [comment]
Dear Filecoin community,
Thank you for your continued patience. The organizers have finished tallying the results and computing the allocations for each project, according to the scoring rules that were previously set forth. Please visit the following resources for the results:

- [FIL-RetroPGF-1 Results overview](https://fil-retropgf.notion.site/FIL-RetroPGF-1-Results-9ad30232b643460598353489d79b9628)
- [Results Spreadsheet](https://docs.google.com/spreadsheets/d/15tQZU4yfzCCso_dd5NGXfSy6XB2pmrEZzZrG_TEIKyY/edit#gid=1610641018)
- [Deep dive into the results.](https://medium.com/p/7e5a0bcdba08)

We will contact project POCs in the coming days to coordinate distribution. A huge thank you to the entire Filecoin ecosystem for coming together to make this first round successful. Be on the lookout for Round 2!

**Comment by kkarrancsu** [comment]
Dear Filecoin Community,

Thank you for your continued patience. We are now in the next phase of FIL-RetroPGF-1—funds distribution. The [Open Impact Foundation](https://openimpact.foundation/) will coordinate distribution and contact projects [that have been allocated funds](https://fil-retropgf.notion.site/FIL-RetroPGF-1-Results-9ad30232b643460598353489d79b9628?pvs=74) from the FIL-RetroPGF-1 process.

Please be on the lookout in your inbox for emails coming from hello@openimpact.foundation

Best Regards,
